### ISSUE TITLE
GetValue assigned List but is expected to return dictionary? Binance (C#)

### ISSUE THREAD

**Issue by JakubMartinovicHusar** [ccxt/ccxt#23806]
### Operating System

Ubuntu 22.04

### CCXT Version

4.3.42

### Description

Hi I am trying to recompile CCXT to .net 8. But as I am running it I am getting errors. That on it's own is expected as current version of CCXT has been compiled on older .net. However I can not understand this piece of code. First line fills value for crossMarginPairsData key with list but on line 28666  output of getValue method is casted to IDictionary<string,object>. What is reason behind this?

Thank you for explanation. Also is there any CCXT version compiled to .NET 8?

Thank you.

### Code

```
  

((IDictionary<string,object>)this.options)["crossMarginPairsData"] = new List<object>() {};
        ((IDictionary<string,object>)this.options)["isolatedMarginPairsData"] = new List<object>() {};
        for (object i = 0; isLessThan(i, getArrayLength(results)); postFixIncrement(ref i))
        {
            object res = this.safeValue(results, i);
            if (isTrue(isTrue(fetchMargins) && isTrue(((res is IList<object>) || (res.GetType().IsGenericType && res.GetType().GetGenericTypeDefinition().IsAssignableFrom(typeof(List<>)))))))
            {
                object keysList = new List<object>(((IDictionary<string,object>)this.indexBy(res, "symbol")).Keys);
                object length = getArrayLength((new List<object>(((IDictionary<string,object>)getValue(this.options, "crossMarginPairsData")).Keys)));
```

**Comment by carlosmiei** [comment]
Hello @JakubMartinovicHusar,
That code is transpiled from a typescript source that's why it seems really odd and illegible, it's easier to check the original code. 

@JakubMartinovicHusar regarding the net8 question I think it should work with no issues (but I might need to double check it)
![image](https://github.com/user-attachments/assets/6d8a0949-c55d-4406-938d-8cd76be15445)

**Comment by carlosmiei** [comment]
@JakubMartinovicHusar btw locally I use dotnet 8 and I can compile it with no issues

```
dotnet --version   
8.0.101
```

**Comment by carlosmiei** [comment]
@JakubMartinovicHusar you're right we have an issue there but will be fixed shortly here: https://github.com/ccxt/ccxt/pull/23808

**Comment by JakubMartinovicHusar** [comment]
Thank you for quick answer. I compiled code to .net 8 as well after fixing
that issue. I am still testing functionalities though. Here is the code.
Feel free to use it if you like it (ccxt/cs/ccxt/exchanges/binance.cs:2867).
object length = getArrayLength(
                    (List<object>)((getValue((IDictionary<string,object>)
this.options, "crossMarginPairsData"))));

<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avg.com
<http://www.avg.com/email-signature?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by carlosmiei** [comment]
@JakubMartinovicHusar Can you please update and try again? this issue should be fixed now
